Fix TestDateParse fail due to ParseDate().
"21 Mar 2006" will be translated as text with Chinese characters on Chinese version windows.
Inside wxStrchr() it will trigger assertion fail because it try to convert non-ascii character to char.

This is because
	wxStrchr(const char *s, const wxUniChar& c)
will call
	wxCRT_StrchrA()

This patch use
	wxStrchr(const wchar_t *s, const wxUniChar& c)
instead.

The other possible fix is change
	wxStrchr(const char *s, const wxUniChar& c)
to call
	wxCRT_StrchrW()
	

diff --git a/wxWidgets/src/common/datetimefmt.cpp b/wxWidgets/src/common/datetimefmt.cpp
--- a/wxWidgets/src/common/datetimefmt.cpp
+++ b/wxWidgets/src/common/datetimefmt.cpp
@@ -1613,17 +1613,17 @@
     wxDateTime_t day = 0;
     wxDateTime::Month mon = Inv_Month;
     int year = 0;
 
     // tokenize the string
     while ( p != pEnd )
     {
         // skip white space and date delimiters
-        while ( wxStrchr(".,/-\t\r\n ", *p) )
+        while ( wxStrchr(wxT(".,/-\t\r\n "), *p) )
         {
             ++p;
         }
 
         // modify copy of the iterator as we're not sure if the next token is
         // still part of the date at all
         wxString::const_iterator pCopy = p;
 
